### PR TITLE
Update paths and add maven build system

### PR DESCRIPTION
**Pull request by dgault**
The aim of this PR is to take the result of git filter-branch in ome-poi, which is currently not buildable, and to make it build with maven and travis. This will get the repository functional so that this basic infrastructure will be in place for future PRs.

- Use standard maven path layout (src files moved from components/forks/poi to src/main/java)
- Drop ant build
- Add travis support for maven build only
- Make maven build work again (NOTE: ome-common dependency changed to use LATEST version)
- Update licence and readme - license used is apache license from components/forks/poi as opposed to bsd license

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,67 +5,37 @@
...
+      <url>http://www.apache.org/licenses/LICENSE-2.0</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.openmicroscopy</groupId>
+      <artifactId>ome-common</artifactId>
+      <version>LATEST</version>
```
Comment (dgault) [inline pom.xml]: Yeah I looked into alternatives and haven't found anything better as of yet, any other suggestions would certainly be welcome.

**Comment by rleigh-codelibre** [comment]
From `mvn dependency:analyze`:

```
WARNING] Used undeclared dependencies found:
[WARNING]    xml-apis:xml-apis:jar:1.3.04:runtime
```

**Comment by rleigh-codelibre** [comment]
From `mvn versions:display-dependency-updates`:

```
[INFO] artifact commons-logging:commons-logging: checking for updates from central
[INFO] artifact org.openmicroscopy:ome-common: checking for updates from central
[INFO] The following dependencies in Dependencies have newer versions:
[INFO]   commons-logging:commons-logging ......................... 1.1.1 -> 1.2****
```

**Comment by rleigh-codelibre** [comment]
This all looks great.  I just noted two minor things maven picked up.  I don't know how bad these are, but it might be worth fixing up so it's clean.

**Comment by sbesson** [comment]
Re `commons-logging`, I think we may want to assess the impact of bumping a library version across the entire Java stack incl. OMERO as there are already many version conflicts resolved for this dependency.

Re `xml-apis`, I remember there are some issues with the output of `dependency:analyze` commented in https://github.com/openmicroscopy/bioformats/blob/v5.2.4/components/formats-api/pom.xml#L51. /cc @melissalinkert

**Comment by rleigh-codelibre** [comment]
@sbesson OK.  If there are any problems we can certainly fix them as followups; this looks good to merge as it is.

**Comment by sbesson** [comment]
Merging then. We can review the 2 points raised by @rleigh-codelibre independently and address them in future version. Thanks all.

**Pull request merged**